Assets/Scripts/BulletSystem/AbstractClass/MechBullet.cs

```diff
@@ -0,0 +1,40 @@
+using UnityEngine;
+
+/*
+ * Abstract bullet class.
+ * Has a damage value, speed, and lifetime.
+ */
+public abstract class MechBullet : MonoBehaviour
```

Comment:
Any reason why this can't use the regular Bullet class? Looks like identical code at a glance.

Reply:
It was because the damage receiver class causes the mech to take damage whenever a Bullet hits it. I could maybe edit the Bullet class to include a source though, and check that to determine if something should take damage or not?

Reply:
> I could maybe edit the Bullet class to include a source though, and check that to determine if something should take damage or not?

That sounds better than duplicating the files, yeah